game-core/src/main/java/games/strategy/engine/framework/GameDataUtils.java

```diff
@@ -15,6 +16,18 @@
 public final class GameDataUtils {
   private GameDataUtils() {}
 
+  /**
+   * Create a deep copy of GameData without history as it can get large.
+   * <strong>You should have the game data's write lock before calling this method</strong>
+   */
+  public static GameData cloneGameDataWithoutHistory(final GameData data, final boolean copyDelegates) {
```

Comment:
Do the delegates also get large? Were you able to gain any insight into why one caller passes false and the other true? I'm wondering if the delegates copy option was maybe for a similar reason.

Reply:
I think so though probably not to the extent that game history does for large maps with many rounds.

Reply:
In theory we could override the writeObject method so we can enable and disable the serialization of different fields based on some sort of boolean value.
Not sure how complicated that would be, but it would be an option.

Reply:
Yeah, that crossed my mind as well but given that delegates were handled this way and figured its better to align the history in a similar manner. Unless you think the writeObject approach would have some benefits?

Reply:
We wouldn't need to aquire any write locks and could keep the field final.

Reply:
Wouldn't you still need a write lock?

Reply:
Not if a potential boolean is volatile or something.

Reply:
I'd have to think about it some. Either way, I'm not going to have much time the next few weeks so I'd say if we think this is a safe/useful change then we merge it. Otherwise if someone wants to try a different approach and close/modify this I'm fine with that.